### PR TITLE
fix Error when creating an account on e2e build

### LINKED ISSUE #20806: Error when creating an account on e2e build
# Bug Report

## Problem 
Sometimes when creating an account we get a popup with error:

Screenshot 2024-07-17 at 19 13 02

#### Expected behavior
No error

#### Actual behavior
Please shake your phone to report this error... is shown

### Reproduction

1. Create a new account

### Additional Information

- Status version: nightly 
- Operating System: Android

Logs:
[logcat_shake_the_phone_error.log](https://github.com/user-attachments/files/16286210/logcat_shake_the_phone_error.log)

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/20806

### Summary
Navigate to next screen once onboarding animation and account creation both are complete.

Note: If account creation fails, user will get stuck on loading screen.

### Testing
Needs to adjust tests as per https://github.com/status-im/status-mobile/pull/21471

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/contexts/onboarding/events.cljs [hunk 1]
```diff
@@ -67,6 +67,28 @@
  (fn [_ [error]]
    {:dispatch [:biometric/show-message (ex-cause error)]}))
 
+;; Navigate to next screen once onboarding animation and account creation both are complete
+;; Note: If account creation fails, user will get stuck on loading screen
+;; TODO: Show error screen in case account creation/restoration fails
+(rf/defn account-creation-complete
```
Comment (ilmotta) [inline src/status_im/contexts/onboarding/events.cljs]: @Parveshdhull we lose linting when this macro is used. If this event is not being used in places with `rf/merge` we should just use `reg-event-fx`.

File: src/status_im/contexts/onboarding/events.cljs [hunk 2]
```diff
@@ -67,6 +67,28 @@
...
+  {:events [:onboarding/account-creation-complete]}
+  [{:keys [db] :as cofx} {:keys [loading-animation-complete? loading-screen login-signal-received?]}]
+  (let [loading-animation-complete? (or loading-animation-complete?
```
Comment (ilmotta) [inline src/status_im/contexts/onboarding/events.cljs]: We had a few discussions in the past about avoiding shadowing, but I'd need to dig to find them. But the decision was to prefer different names. Destructuring is what's causing the problem in this case. So an alternative that fixes shadowing:

```clojure
(rf/reg-event-fx :onboarding/account-creation-complete
 (fn [{:keys [db]} {:keys [opts]}]
   (let [loading-animation-complete? (or (:loading-animation-complete? opts)
                                         (:onboarding/loading-animation-complete? db))
         loading-screen              (or (:loading-screen opts) (:onboarding/loading-screen db))
         login-signal-received?      (or (:login-signal-received? opts)
                                         (:onboarding/login-signal-received? db))]
     (if (and loading-animation-complete? login-signal-received?)
       {:db (-> db
                (dissoc :onboarding/loading-screen)
                (dissoc :onboarding/loading-animation-complete?)
                (dissoc :onboarding/login-signal-received?))
        :fx [[:dispatch
              [:navigate-to-within-stack [:screen/onboarding.enable-notifications loading-screen]]]]}
       {:db (-> db
                (assoc :onboarding/loading-screen loading-screen)
                (assoc :onboarding/loading-animation-complete? loading-animation-complete?)
                (assoc :onboarding/login-signal-received? login-signal-received?))}))))
```

**Review by ilmotta — APPROVED** [review]

File: src/status_im/contexts/onboarding/events.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im/contexts/onboarding/events.cljs]: Thank you for reviewing the PR, updated

**Comment by Parveshdhull** [comment]
Hi! If we improve our onboarding process, we might also consider improving the animation delay. Here’s how the onboarding currently looks on my emulator with and without the [animation delay](https://github.com/status-im/status-mobile/blob/020d5cc1c124f9bd2e8931d5275102a7b093d573/src/status_im/constants.cljs#L453). (probably more faster on some devices)

https://github.com/user-attachments/assets/4938968f-ca9f-4951-91be-affe1d57333c

cc: @ilmotta @churik @flexsurfer

**Comment by Parveshdhull** [comment]
By the way, this isn’t critical, but I just wanted to raise this point. To me, It seems a bit inconsistent that in some areas we’re focused on saving milliseconds for performance, while in others we’re introducing delays of few seconds.

**Comment by ilmotta** [comment]
> By the way, this isn’t critical, but I just wanted to raise this point. To me, It seems a bit inconsistent that in some areas we’re focused on saving milliseconds for performance, while in others we’re introducing delays of few seconds.

I took note of your feedback @Parveshdhull to consider in the mini-epic to simplify onboarding for the 2.32 release. There are still some details to align with designers and PL.

**Comment by churik** [comment]
So I made 5 rounds of e2e in https://github.com/status-im/status-mobile/pull/21471 - I didn't face this issue. 

And no regression from the PR as well.

I tested briefly account creation and sync on IOS - it works. We should merge and cherry pick this to the release branch, exceptional work @Parveshdhull - thank you!

**Comment by churik** [comment]
this PR closes https://github.com/status-im/status-mobile/issues/21151 as well (hopefully!!!)

**Pull request merged**